nx-X11/programs/Xserver/hw/nxagent/Dialog.c

```diff
@@ -315,13 +315,23 @@ void nxagentLaunchDialog(DialogType dialogType)
   fprintf(stderr, "nxagentLaunchDialog: Launching dialog type [%d] message [%s].\n", type, message);
   #endif
 
+  char *dialogDisplay = NULL;
+  int len = 0;
   if (dialogType == DIALOG_FAILED_RECONNECTION)
   {
-    snprintf(dialogDisplay, sizeof(dialogDisplay), "%s", nxagentDisplayName);
+    len = asprintf(&dialogDisplay, "%s", nxagentDisplayName);
   }
   else
   {
-    snprintf(dialogDisplay, sizeof(dialogDisplay), ":%s", display);
+    len = asprintf(&dialogDisplay, ":%s", display);
+  }
+
+  if (len == -1)
+  {
+    #ifdef DEBUG
+    fprintf(stderr, "%s: could not allocate display string.\n", __func__);
+    #endif
+    return;
```

Comment:
So... we simply ignore memory allocation errors here? Is that intended?

Reply:
well, what else are you suggesting?

Reply:
hmmm... ok...